### PR TITLE
Add csv slicer

### PR DESCRIPTION
**Pull request by SgtMarmite**
https://keboola.atlassian.net/browse/GCP-510
https://keboola.atlassian.net/browse/GCP-511

- added csv slicer for files bigger than 1GB
- retry refactor
- added Threading for chunk sending

Tested by @ZdenekSrotyr in client's project

### CODE REVIEW

File: google_bigquery_writer/writer.py [hunk 1]
```diff
@@ -196,3 +209,30 @@ def write_table_sync(
...
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+
+            if result.returncode != 0:
+                raise UserException(f"Error running cli_linux_amd64: {result.stderr.strip()}")
+
+        except Exception as e:
+            raise UserException(f"Error running cli_linux_amd64: {e}")
```
Comment (davidesner) [inline google_bigquery_writer/writer.py]: I would keep that as App error. The user has probably no way of fixing it so we should know about it.

**Review by davidesner — CHANGES_REQUESTED** [review]
One minor comment

**Comment by SgtMarmite** [comment]
> One minor comment

updated

**Review by davidesner — APPROVED** [review]
lgtm

**Pull request merged**